client/clientutil/snapinfo_test.go

```diff
@@ -0,0 +1,278 @@
+			Username:    "thingyinc",
+			DisplayName: "Thingy Inc.",
+			Validation:  "unproven",
+		},
+		Base: "core18",
+		SideInfo: snap.SideInfo{
+			RealName:          "the-snap",
+			SnapID:            "snapidid",
+			Revision:          snap.R(99),
+			EditedTitle:       "the-title",
+			EditedSummary:     "the-summary",
+			EditedDescription: "the-description",
+			Channel:           "latest/stable",
+			Contact:           "https://thingy.com",
+			Private:           true,
+		},
+		Channels: map[string]*snap.ChannelSnapInfo{},
+		Tracks:   []string{},
+		Prices:   map[string]float64{},
+		Media: []snap.MediaInfo{
+			{Type: "icon", URL: "https://dashboard.snapcraft.io/site_media/appmedia/2017/12/Thingy.png"},
+			{Type: "screenshot", URL: "https://dashboard.snapcraft.io/site_media/appmedia/2018/01/Thingy_01.png"},
+			{Type: "screenshot", URL: "https://dashboard.snapcraft.io/site_media/appmedia/2018/01/Thingy_02.png", Width: 600, Height: 200},
+		},
+		CommonIDs: []string{"org.thingy"},
+		Website:   "http://example.com/thingy",
+		StoreURL:  "https://snapcraft.io/thingy",
+		Broken:    "broken",
+	}
+	// valid InstallDate
+	err := os.MkdirAll(si.MountDir(), 0755)
+	c.Assert(err, IsNil)
+	err = os.Symlink(si.Revision.String(), filepath.Join(filepath.Dir(si.MountDir()), "current"))
+	c.Assert(err, IsNil)
+
+	ci, err := clientutil.ClientSnapFromSnapInfo(si, nil)
+	c.Check(err, IsNil)
+
+	// check that fields are filled
+	// see daemon/snap.go for fields filled after this
+	expectedZeroFields := []string{
+		"Screenshots", // unused nowadays
+		"DownloadSize",
+		"InstalledSize",
+		"Health",
+		"Status",
+		"TrackingChannel",
+		"IgnoreValidation",
+		"CohortKey",
+		"DevMode",
+		"TryMode",
+		"JailMode",
+		"MountedFrom",
+	}
+	var checker func(string, reflect.Value)
+	checker = func(pfx string, x reflect.Value) {
+		t := x.Type()
+		for i := 0; i < x.NumField(); i++ {
+			f := t.Field(i)
+			if f.PkgPath != "" {
+				// not exported, ignore
+				continue
+			}
+			v := x.Field(i)
+			if f.Anonymous {
+				checker(pfx+f.Name+".", v)
+				continue
+			}
+			if reflect.DeepEqual(v.Interface(), reflect.Zero(f.Type).Interface()) {
+				name := pfx + f.Name
+				c.Check(expectedZeroFields, testutil.Contains, name, Commentf("%s not set", name))
+			}
+		}
+	}
+	x := reflect.ValueOf(ci).Elem()
+	checker("", x)
+
+	// check some values
+	c.Check(ci.Name, Equals, "the-snap_insta")
+	c.Check(ci.Type, Equals, "app")
+	c.Check(ci.ID, Equals, si.ID())
+	c.Check(ci.Revision, Equals, snap.R(99))
+	c.Check(ci.Version, Equals, "v1")
+	c.Check(ci.Title, Equals, "the-title")
+	c.Check(ci.Summary, Equals, "the-summary")
+	c.Check(ci.Description, Equals, "the-description")
+	c.Check(ci.Icon, Equals, si.Media.IconURL())
+	c.Check(ci.Website, Equals, si.Website)
+	c.Check(ci.StoreURL, Equals, si.StoreURL)
+	c.Check(ci.Developer, Equals, "thingyinc")
+	c.Check(ci.Publisher, DeepEquals, &si.Publisher)
+}
+
+type testStatusDecorator struct{}
```

Comment:
Should we add something like:
```
-type testStatusDecorator struct{}
+type testStatusDecorator struct {
+       calls int
+}
```
and do "calls++" in testStatusDecorator.DecorateWithStatus so that this can be checked in the tests and we are explicit about it?